deployment/lib/cf-url-rewriter.js

```diff
@@ -0,0 +1,59 @@
+const aws = require('aws-sdk');
+
+const s3 = new aws.S3({ apiVersion: '2006-03-01' });
+
+exports.handler = async (event, context) => {
+
+    const request = event.Records[0].cf.request;
+    request.uri = request.uri.replace(/^\/ci\/...\//, '\/')
+    callback(null, request);
+
+    // Below is the working in progress logic to get the max build number under a version.
+    // Hardcode the version to be 1.2.0 and bucket name to test-access-1-20 for demo purpose.
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    // Get the object from the event and show its content type
+    const bucket = "test-access-1-20";
+
+    // Create the parameters for calling listObjects
+    var bucketParams = {
+        Bucket: bucket,
+        Prefix: '1.2.0/',
+        Delimiter: '/'
+    };
+
+    try {
+        const s3Response = await s3.listObjects(bucketParams).promise();
```

Comment:
I would invert the flow of control so that you are only checking s3 when you know that you need to look up a wild card.  Psuedo code could look like:

```javascript
if (!request.uri.match('/latest/')) {
   // Do no more processing if the keyword isn't found
   callback(null, request);
} else {
   let listedWholeBucket = false;
   do {
   const requestRoot = request.uri.substring(0, request.uri.indexOf('/latest/'));
   const possibleVersions = s3.listObjects({
      bucket: process.env.bucketName,
      prefix: requestRoot,
      ...
      });
  } while (!possibleVersions.IsTruncated);
  const lexigraphicalLastItem = possibleVersions.Contents[possibleVersions.length -1];
  request.uri = request.uri.replace('/latest/', "/" + lexigraphicalLastItem['Prefix'] + "/")
  callback(null, request);
}
```

Reply:
Please use the method described by @dblock https://github.com/opensearch-project/opensearch-build/pull/1569#discussion_r795250044 for the actual query instead of 'deep page, instead of the truncated bucket unwrapping as suggest here

Reply:
Switched to read from `index.json` file. The s3 part is no longer applicable. https://github.com/opensearch-project/opensearch-build/pull/1622